### ISSUE TITLE
Control over need lost after  client cerfificate deletion

### ISSUE THREAD

**Issue by heikofriedrich** [researchstudio-sat/webofneeds#1371]
During #1335 the folder won-client-certs on satvm01 was deleted after the matchat.org certificate was expanded with the UKI domain names. After that (it seems) old needs that were created before that event cannot be closed anymore or cannot accept requests. Old needs however can still be viewed in the account. The access to them should be totally restricted if the right certificates are not available anymore (also no viewing)!? What happened?
The exception in the owner application log is:

````
8.1.18 16:48:54.601 [http-apr-8443-exec-21] WARN  w.c.service.KeyStoreService - No certificate found for alias https://node.matchat.org/won/resource/need/3554928326595242000
8.1.18 16:48:54.603 [http-apr-8443-exec-21] ERROR w.p.m.p.i.SignatureAddingWonMessageProcessor - Failed to sign
java.lang.NullPointerException: null
        at won.cryptography.rdfsign.WonSigner.sign(WonSigner.java:87) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.cryptography.rdfsign.WonSigner.sign(WonSigner.java:118) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.protocol.message.processor.impl.WonMessageSignerVerifier.signContents(WonMessageSignerVerifier.java:112) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.protocol.message.processor.impl.WonMessageSignerVerifier.sign(WonMessageSignerVerifier.java:38) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.protocol.message.processor.impl.SignatureAddingWonMessageProcessor.processWithKey(SignatureAddingWonMessageProcessor.java:59) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.protocol.message.processor.impl.SignatureAddingWonMessageProcessor.processOnBehalfOfNeed(SignatureAddingWonMessageProcessor.java:50) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.owner.messaging.OwnerWonMessageSenderJMSBased.doSigningOnOwner(OwnerWonMessageSenderJMSBased.java:122) [won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.messaging.OwnerWonMessageSenderJMSBased.sendWonMessage(OwnerWonMessageSenderJMSBased.java:74) [won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.service.impl.OwnerApplicationService.sendWonMessage(OwnerApplicationService.java:36) [won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.web.websocket.WonWebSocketHandler.handleTextMessage(WonWebSocketHandler.java:177) [classes/:na]
        at org.springframework.web.socket.handler.AbstractWebSocketHandler.handleMessage(AbstractWebSocketHandler.java:43) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.WebSocketHandlerDecorator.handleMessage(WebSocketHandlerDecorator.java:75) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.LoggingWebSocketHandlerDecorator.handleMessage(LoggingWebSocketHandlerDecorator.java:56) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.ExceptionWebSocketHandlerDecorator.handleMessage(ExceptionWebSocketHandlerDecorator.java:58) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.session.AbstractSockJsSession.delegateMessages(AbstractSockJsSession.java:382) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.session.WebSocketServerSockJsSession.handleMessage(WebSocketServerSockJsSession.java:193) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.handler.SockJsWebSocketHandler.handleTextMessage(SockJsWebSocketHandler.java:92) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.AbstractWebSocketHandler.handleMessage(AbstractWebSocketHandler.java:43) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter.handleTextMessage(StandardWebSocketHandlerAdapter.java:110) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter.access$000(StandardWebSocketHandlerAdapter.java:42) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter$3.onMessage(StandardWebSocketHandlerAdapter.java:81) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter$3.onMessage(StandardWebSocketHandlerAdapter.java:78) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.apache.tomcat.websocket.WsFrameBase.sendMessageText(WsFrameBase.java:393) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processDataText(WsFrameBase.java:494) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processData(WsFrameBase.java:289) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processInputBuffer(WsFrameBase.java:130) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.server.WsFrameServer.onDataAvailable(WsFrameServer.java:60) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.server.WsHttpUpgradeHandler$WsReadListener.onDataAvailable(WsHttpUpgradeHandler.java:183) [tomcat-websocket.jar:8.0.30]
        at org.apache.coyote.http11.upgrade.AbstractServletInputStream.onDataAvailable(AbstractServletInputStream.java:198) [tomcat-coyote.jar:8.0.30]
        at org.apache.coyote.http11.upgrade.AbstractProcessor.upgradeDispatch(AbstractProcessor.java:96) [tomcat-coyote.jar:8.0.30]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:669) [tomcat-coyote.jar:8.0.30]
        at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2500) [tomcat-coyote.jar:8.0.30]
        at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2489) [tomcat-coyote.jar:8.0.30]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.0.30]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
8.1.18 16:48:54.603 [http-apr-8443-exec-21] INFO  w.o.s.impl.OwnerApplicationService - could not send WonMessage
java.lang.RuntimeException: could not send message
        at won.owner.messaging.OwnerWonMessageSenderJMSBased.sendWonMessage(OwnerWonMessageSenderJMSBased.java:103) ~[won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.service.impl.OwnerApplicationService.sendWonMessage(OwnerApplicationService.java:36) ~[won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.web.websocket.WonWebSocketHandler.handleTextMessage(WonWebSocketHandler.java:177) [classes/:na]
        at org.springframework.web.socket.handler.AbstractWebSocketHandler.handleMessage(AbstractWebSocketHandler.java:43) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.WebSocketHandlerDecorator.handleMessage(WebSocketHandlerDecorator.java:75) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.LoggingWebSocketHandlerDecorator.handleMessage(LoggingWebSocketHandlerDecorator.java:56) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.ExceptionWebSocketHandlerDecorator.handleMessage(ExceptionWebSocketHandlerDecorator.java:58) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.session.AbstractSockJsSession.delegateMessages(AbstractSockJsSession.java:382) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.session.WebSocketServerSockJsSession.handleMessage(WebSocketServerSockJsSession.java:193) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.sockjs.transport.handler.SockJsWebSocketHandler.handleTextMessage(SockJsWebSocketHandler.java:92) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.handler.AbstractWebSocketHandler.handleMessage(AbstractWebSocketHandler.java:43) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter.handleTextMessage(StandardWebSocketHandlerAdapter.java:110) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter.access$000(StandardWebSocketHandlerAdapter.java:42) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter$3.onMessage(StandardWebSocketHandlerAdapter.java:81) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.web.socket.adapter.standard.StandardWebSocketHandlerAdapter$3.onMessage(StandardWebSocketHandlerAdapter.java:78) [spring-websocket-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.apache.tomcat.websocket.WsFrameBase.sendMessageText(WsFrameBase.java:393) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processDataText(WsFrameBase.java:494) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processData(WsFrameBase.java:289) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.WsFrameBase.processInputBuffer(WsFrameBase.java:130) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.server.WsFrameServer.onDataAvailable(WsFrameServer.java:60) [tomcat-websocket.jar:8.0.30]
        at org.apache.tomcat.websocket.server.WsHttpUpgradeHandler$WsReadListener.onDataAvailable(WsHttpUpgradeHandler.java:183) [tomcat-websocket.jar:8.0.30]
        at org.apache.coyote.http11.upgrade.AbstractServletInputStream.onDataAvailable(AbstractServletInputStream.java:198) [tomcat-coyote.jar:8.0.30]
        at org.apache.coyote.http11.upgrade.AbstractProcessor.upgradeDispatch(AbstractProcessor.java:96) [tomcat-coyote.jar:8.0.30]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:669) [tomcat-coyote.jar:8.0.30]
        at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2500) [tomcat-coyote.jar:8.0.30]
        at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2489) [tomcat-coyote.jar:8.0.30]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.0.30]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
Caused by: won.protocol.message.processor.exception.WonMessageProcessingException: Failed to sign message https://node.matchat.org/won/resource/event/2939095694777190400
        at won.protocol.message.processor.impl.SignatureAddingWonMessageProcessor.processOnBehalfOfNeed(SignatureAddingWonMessageProcessor.java:53) ~[won-cryptography-0.2-SNAPSHOT.jar:na]
        at won.owner.messaging.OwnerWonMessageSenderJMSBased.doSigningOnOwner(OwnerWonMessageSenderJMSBased.java:122) ~[won-owner-0.2-SNAPSHOT.jar:na]
        at won.owner.messaging.OwnerWonMessageSenderJMSBased.sendWonMessage(OwnerWonMessageSenderJMSBased.java:74) ~[won-owner-0.2-SNAPSHOT.jar:na]
        ... 29 common frames omitted

````

**Comment by heikofriedrich** [comment]
Even with the security update of commit 03f1a1812e92c32e3ffbc46a22bb1c7b0f26d567 viewing the content (of closed needs on matchat) is still possible !!

**Comment by fkleedorfer** [comment]
The conversation still being accessible turned out to be a bug in our spring-security config - fixed as of  #1372. The keystore file as a single point of failure has been eliminated in  #1392.
